### ISSUE TITLE
Subcommands do not fallback to app.ExitEventHandler

### ISSUE THREAD

**Issue by AlexeiSosnov** [urfave/cli#816]
When I install ExitEventHandler on the app level and exit a subcommand with cli.NewExitError then ExitEventHandler is not called, and the app exits through OsExiter unexpectedly.

It is because original app.ExitEventHandler is not inherited to the app copy in command.go startApp function

A test to illustrate it:

```golang
func TestWhenExitSubCommandWithCodeThenAppQuitUnexpectedly(t *testing.T) {
	app := cli.NewApp()
	app.Commands = []cli.Command{
		cli.Command{
			Name: "cmd",
			Subcommands: []cli.Command{
				cli.Command{
					Name: "subcmd",
					Action: func(c *cli.Context) error {
						return cli.NewExitError("exit error", 101)
					},
				},
			},
		},
	}

	origExiter := cli.OsExiter
	var unhandledExitCode int
	cli.OsExiter = func(exitCode int) {
		unhandledExitCode = exitCode
	}

	var handledExitCode int
	app.ExitErrHandler = func(c *cli.Context, err error) {
		if exitErr, ok := err.(cli.ExitCoder); ok {
			handledExitCode = exitErr.ExitCode()
		}
	}

	app.Run([]string{
		"myapp",
		"cmd",
		"subcmd",
	})

	if unhandledExitCode != 101 {
		t.Errorf("unhandled exit code %v is handled for subcommand", unhandledExitCode)
	}

	if handledExitCode != 0 {
		t.Errorf("handled code %v is handled for subcommand", handledExitCode)
	}

	cli.OsExiter = origExiter
}
```

**Comment by lynncyrin** [comment]
I haven't looked at this issue in depth, but it makes sense! Would you be interested in creating a PR for it @AlexeiSosnov?

**Comment by FaranIdo** [comment]
I think this issue can be closed, solved in commit 08c24e22ed2c4bebb348a738caf92c40bb63133c (pull request #856 )
